### PR TITLE
[Workflow] Refresh edit UI

### PR DESCRIPTION
**Pull request by zerolab**
This PR adds UI changes to the edit page (and minor improvements to the homepage) needed for Workflow

Notes:
- this includes the changes from #6151 (to avoid duplicating work). 
- the PR builds on #6084 and so it targets the audit log branch to reduce the cognitive load on review

Screenshots/screencast coming shortly

* Do the tests still pass? (https://docs.wagtail.io/en/latest/contributing/developing.html#testing)
* Does the code comply with the style guide? (Run `make lint` from the Wagtail root)
* For Python changes: Have you added tests to cover the new/fixed behaviour?
* For front-end changes: Did you test on all of Wagtail’s supported browsers? **Please list the exact versions you tested**.
* For new features: Has the documentation been updated accordingly?

### CODE REVIEW

File: client/scss/components/_button.scss [hunk 1]
```diff
@@ -124,6 +126,37 @@
...
+            line-height: 1.85em;
+            height: 100%;
+            text-align: center;
+            border-top-left-radius: inherit;
+            border-bottom-left-radius: inherit;
+        }
+
+        &.button--icon {
+            &:before {
+                display: none;  // TODO: remove once the icon font styles are gone
```
[the remark this answers is not in the document]
Comment (zerolab) [reply client/scss/components/_button.scss:145]: Yes. Once #6107 is done and we deprecate the use of the font icon, we can remove this.
This reminds me, these styles come from #6151 so if you could give that one a look, we can make this PR smaller

File: client/scss/components/_dropdown.scss [hunk 2]
```diff
@@ -23,6 +23,30 @@
     display: inline-block;
 }
 
+.c-dropdown__togle--icon {
+    &:before {
+        display: none; // TODO: remove when iconfont styles are removed
```
[the remark this answers is not in the document]
Comment (zerolab) [reply client/scss/components/_dropdown.scss:28]: Same as above

File: wagtail/contrib/redirects/templates/wagtailredirects/index.html [hunk 3]
```diff
@@ -1,6 +1,10 @@
 {% extends "wagtailadmin/base.html" %}
 {% load i18n %}
+<<<<<<< HEAD
```
[the remark this answers is not in the document]
Comment (zerolab) [reply wagtail/contrib/redirects/templates/wagtailredirects/index.html]: Eagle eyes!

File: wagtail/admin/action_menu.py [hunk 4]
```diff
@@ -248,10 +249,16 @@ def __init__(self, request, **kwargs):
...
             task = page.current_workflow_task
+            is_final_task = page.current_workflow_state and page.current_workflow_state.is_at_final_task
             if task:
                 actions = task.get_actions(page, request.user)
-                workflow_menu_items = [WorkflowMenuItem(name, label, launch_modal) for name, label, launch_modal in actions]
-                workflow_menu_items = [item for item in workflow_menu_items if item.is_shown(self.request, self.context)]
+                workflow_menu_items = []
+                for name, label, launch_modal in actions:
+                    if name == "approve" and is_final_task:
+                        label = format_lazy(_("{label} and Publish"), label=label)
```
Comment (gasman) [inline wagtail/admin/action_menu.py]: We should use old-style string interpolation here - Transifex doesn't recognise this style of format string, and will not guard against translators trying to translate the word "label" (which will cause an error for that one language).

File: wagtail/admin/action_menu.py [hunk 5]
```diff
@@ -20,6 +20,7 @@ class ActionMenuItem(metaclass=MediaDefiningClass):
     label = ''
     name = None
     classname = ''
+    icon_name = ''
```
Comment (gasman) [inline wagtail/admin/action_menu.py:23]: This should be added to the documentation at https://github.com/wagtail/wagtail/blob/master/docs/reference/hooks.rst#register_page_action_menu_item

File: wagtail/admin/action_menu.py (hunk 4 above)
Comment (gasman) [reply wagtail/admin/action_menu.py]: There are a bunch of these in register_workflow_log_actions too, so will do that en masse in a follow-up PR

**Comment by gasman** [comment]
Merged into feature/workflow in a4b35dc600d1f0836b5a8b095cf3fa0c4760ef35.

**Pull request closed without merging**